### PR TITLE
JAMES 3055  Update Deleted Messages Vault

### PR DESCRIPTION
**Pull request by naviat**

### CODE REVIEW

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 1]
```diff
@@ -493,3 +494,41 @@ Managing an email queue is an easy task if you follow this procedure:
...
+If the Deleted Messages Vault is enabled when users delete their mails, and by that we mean when they try to definitely delete them by emptying the trash, James will retain these mails into the Deleted Messages Vault, before an email or a mailbox is going to be deleted. And only administrators can interact with this component via [WebAdmin REST APIs](manage-webadmin.html#deleted-messages-vault).
+
+However, mails are not retained forever as you have to configure a retention period before using it (with one-year retention by default if not defined). It's also possible to permanently delete an mail if needed.
+
+### How to recovering deleted messages
+
+To setup James with Deleted Messages Vault, you need to following this step:
+
+- Enable Deleted Messages Vault by configuring Pre Deletion Hooks.
+- Configuring your BlobStore.
```
Comment (chibenwa) [inline src/site/markdown/server/manage-guice-distributed-james.md]: With that product we always have a blob store, please remove this line.

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 2]
```diff
@@ -493,3 +494,41 @@ Managing an email queue is an easy task if you follow this procedure:
...
+- Configuring the retention time for the Deleted Messages Vault.
+
+#### Enable Deleted Messages Vault by configuring Pre Deletion Hooks
+
+By default, you need to configure a Pre Deletion Hook to let James use it. Before deleting a mail in James, `PreDeletionHooks` will be triggered to execute all declared hooks. If all hook executions success, then James will process to delete that mail. There is already a `DeletedMessageVaultHook` in James, its job is to store deleted mails into Deleted Messages Vault. Thus, you need to configure this hook in [listeners.xml](https://github.com/apache/james-project/blob/master/dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/listeners.xml) configuration file.
```
Comment (chibenwa) [inline src/site/markdown/server/manage-guice-distributed-james.md]: Point to http://james.apache.org/server/config-listeners.html

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 3]
```diff
@@ -493,3 +494,41 @@ Managing an email queue is an easy task if you follow this procedure:
...
+
+#### Configuring your BlobStore
```
Comment (chibenwa) [inline src/site/markdown/server/manage-guice-distributed-james.md]: This section can be removed IMO

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 4]
```diff
@@ -493,3 +494,41 @@ Managing an email queue is an easy task if you follow this procedure:
...
+
+#### Configuring the retention time for the Deleted Messages Vault
+
+To do this, you have to create a configuration file [deletedMessageVault.properties], then put it into the conf directory of James. There is only one available property you may want to configure at the moment:
+
+- `retentionPeriod`: represent for the period deleted messages allowed to be stored in Deleted Messages Vault (default of one year).
+
+### Cleaning DeletedMessages passed the rentention period
+
+You can delete all deleted messages older than the configured `retentionPeriod` by using `Purge Deleted Messages` API from [web admin documentation](manage-webadmin.html#deleted-messages-vault). By default, this API will run in CRON at 1st day each month.
```
Comment (chibenwa) [inline src/site/markdown/server/manage-guice-distributed-james.md]: ` By default, this API will run in CRON at 1st day each month.` no :trollface: 

We recommand to call this API in CRON at 1st day each month.

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 5]
```diff
@@ -493,3 +494,41 @@ Managing an email queue is an easy task if you follow this procedure:
...
+
+#### Configuring your BlobStore
+
+The Deleted Messages Vault is using a BlobStore to store and manage the deleted messages. A BlobStore is a dedicated component to store blobs, non-indexable content. There are different implementations available for the BlobStore on top of Cassandra or file object storage services like Openstack Swift and AWS S3, you need to configure it in [blob.properties](https://github.com/apache/james-project/blob/master/dockerfiles/packaging/guice/cassandra/package/etc/james/templates/blobstore.properties) (following this [guide](config-blobstore.html)).
+
+#### Configuring the retention time for the Deleted Messages Vault
+
+To do this, you have to create a configuration file [deletedMessageVault.properties], then put it into the conf directory of James. There is only one available property you may want to configure at the moment:
+
+- `retentionPeriod`: represent for the period deleted messages allowed to be stored in Deleted Messages Vault (default of one year).
```
Comment (chibenwa) [inline src/site/markdown/server/manage-guice-distributed-james.md]: Don't duplicate the content of other documentation files, point to them :-)

Point us to the documentation explaning how to do that: http://james.apache.org/server/config-vault.html

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 6]
```diff
@@ -493,3 +494,45 @@ Managing an email queue is an easy task if you follow this procedure:
...
+
+To setup James with Deleted Messages Vault, you need to following this step:
+
+- Enable Deleted Messages Vault by configuring Pre Deletion Hooks.
+- Configuring your BlobStore.
+- Configuring the retention time for the Deleted Messages Vault.
+
+#### Enable Deleted Messages Vault by configuring Pre Deletion Hooks
+
+By default, you need to configure a Pre Deletion Hook to let James use it. Before deleting a mail in James, `PreDeletionHooks` will be triggered to execute all declared hooks. If all hook executions success, then James will process to delete that mail. There is already a `DeletedMessageVaultHook` in James, its job is to store deleted mails into Deleted Messages Vault. Thus, you need to configure this hook in [listeners.xml](https://github.com/apache/james-project/blob/master/dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/listeners.xml) configuration file.
```
Comment (Arsnael) [inline src/site/markdown/server/manage-guice-distributed-james.md]: s/If all hook executions success/If all hooks execute successfully

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 7]
```diff
@@ -493,3 +494,45 @@ Managing an email queue is an easy task if you follow this procedure:
...
+
+The Deleted Messages Vault is using a BlobStore to store and manage the deleted messages. A BlobStore is a dedicated component to store blobs, non-indexable content. There are different implementations available for the BlobStore on top of Cassandra or file object storage services like Openstack Swift and AWS S3, you need to configure it in [blob.properties](https://github.com/apache/james-project/blob/master/dockerfiles/packaging/guice/cassandra/package/etc/james/templates/blobstore.properties) (following this [guide](config-blobstore.html)).
+
+#### Configuring the retention time for the Deleted Messages Vault
+
+To do this, you have to create a configuration file [deletedMessageVault.properties], then put it into the conf directory of James. There is only one available property you may want to configure at the moment:
+
+- `retentionPeriod`: represent for the period deleted messages allowed to be stored in Deleted Messages Vault (default of one year).
+
+### Restore deleted messages after deleted
```
Comment (Arsnael) [inline src/site/markdown/server/manage-guice-distributed-james.md]: s/deleted/deletion

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 8]
```diff
@@ -493,3 +494,45 @@ Managing an email queue is an easy task if you follow this procedure:
...
+
+After users deleted their mails and empty emptying the trash, admin can use `Restore Deleted Messages` API from [Admin document](manage-webadmin.html#deleted-messages-vault) to restore all the deleted mails.
```
Comment (Arsnael) [inline src/site/markdown/server/manage-guice-distributed-james.md]: s/empty emtying/emptied

Comment (Arsnael) [reply src/site/markdown/server/manage-guice-distributed-james.md]: Just keep `Restore Deleted Messages` and point the doc directly on it, like:
`[Restore Deleted Messages](manage-webadmin.html#deleted-messages-vault)`

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 9]
```diff
@@ -493,3 +494,45 @@ Managing an email queue is an easy task if you follow this procedure:
...
+
+To do this, you have to create a configuration file [deletedMessageVault.properties], then put it into the conf directory of James. There is only one available property you may want to configure at the moment:
+
+- `retentionPeriod`: represent for the period deleted messages allowed to be stored in Deleted Messages Vault (default of one year).
+
+### Restore deleted messages after deleted
+
+After users deleted their mails and empty emptying the trash, admin can use `Restore Deleted Messages` API from [Admin document](manage-webadmin.html#deleted-messages-vault) to restore all the deleted mails.  
+
+### Cleaning DeletedMessages passed the rentention period
```
Comment (Arsnael) [inline src/site/markdown/server/manage-guice-distributed-james.md]: `Cleaning expired deleted messages` ?

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 10]
```diff
@@ -493,3 +494,45 @@ Managing an email queue is an easy task if you follow this procedure:
...
+
+You can delete all deleted messages older than the configured `retentionPeriod` by using `Purge Deleted Messages` API from [web admin document](manage-webadmin.html#deleted-messages-vault). By default, this API will run in CRON at 1st day each month.
```
Comment (Arsnael) [inline src/site/markdown/server/manage-guice-distributed-james.md]: Same comment as before:
`[Purge Deleted Messages](manage-webadmin.html#deleted-messages-vault)`

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 11]
```diff
@@ -512,27 +512,20 @@ However, mails are not retained forever as you have to configure a retention per
...
-
-#### Configuring your BlobStore
-
-The Deleted Messages Vault is using a BlobStore to store and manage the deleted messages. A BlobStore is a dedicated component to store blobs, non-indexable content. There are different implementations available for the BlobStore on top of Cassandra or file object storage services like Openstack Swift and AWS S3, you need to configure it in [blob.properties](https://github.com/apache/james-project/blob/master/dockerfiles/packaging/guice/cassandra/package/etc/james/templates/blobstore.properties) (following this [guide](config-blobstore.html)).
+You need to configure this hook in [listeners.xml](https://github.com/apache/james-project/blob/master/dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/listeners.xml) configuration file. More details about configuration & example is at [Pre Deletion Hook Configuration](http://james.apache.org/server/config-listeners.html)
```
Comment (Arsnael) [inline src/site/markdown/server/manage-guice-distributed-james.md]: s/is at/can be found at

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 12]
```diff
@@ -493,3 +494,38 @@ Managing an email queue is an easy task if you follow this procedure:
 In case, you need to clear an email queue because there are only spam or trash emails in the email queue you have this procedure to follow:
 
 - All mails from the given mail queue will be deleted with [Clearing a mail queue](manage-webadmin.html#Clearing_a_mail_queue).
+
+## Deleted Messages Vault
+
+Deleted Messages Vault is an interesting feature that will help James users have a chance to:
```
Comment (rouazana) [inline src/site/markdown/server/manage-guice-distributed-james.md]: other features are not interesting? :trollface:

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 13]
```diff
@@ -493,3 +494,38 @@ Managing an email queue is an easy task if you follow this procedure:
...
+- restore & export deleted messages by various criteria.
+- permanently delete some retained messages.
+
+If the Deleted Messages Vault is enabled when users delete their mails, and by that we mean when they try to definitely delete them by emptying the trash, James will retain these mails into the Deleted Messages Vault, before an email or a mailbox is going to be deleted. And only administrators can interact with this component via [WebAdmin REST APIs](manage-webadmin.html#deleted-messages-vault).
+
+However, mails are not retained forever as you have to configure a retention period before using it (with one-year retention by default if not defined). It's also possible to permanently delete a mail if needed.
+
+### How to configure deleted messages vault
+
+To setup James with Deleted Messages Vault, you need to following those steps:
```
Comment (rouazana) [inline src/site/markdown/server/manage-guice-distributed-james.md]: s/following/follow/ ?

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 14]
```diff
@@ -493,3 +494,38 @@ Managing an email queue is an easy task if you follow this procedure:
...
+- Enable Deleted Messages Vault by configuring Pre Deletion Hooks.
+- Configuring the retention time for the Deleted Messages Vault.
+
+#### Enable Deleted Messages Vault by configuring Pre Deletion Hooks
+
+You need to configure this hook in [listeners.xml](https://github.com/apache/james-project/blob/master/dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/listeners.xml) configuration file. More details about configuration & example can be found at [Pre Deletion Hook Configuration](http://james.apache.org/server/config-listeners.html)
+
+#### Configuring the retention time for the Deleted Messages Vault
+
+In order to configure the retention time for the Deleted Messages Vault, an administrator needs to perform fine configuration tunning as explained in [deletedMessageVault.properties](https://github.com/apache/james-project/blob/master/dockerfiles/run/guice/cassandra/destination/conf/deletedMessageVault.properties). Mails are not retained forever as you have to configure a retention period (by `retentionPeriod`) before using it (with one-year retention by default if not defined)
```
Comment (rouazana) [inline src/site/markdown/server/manage-guice-distributed-james.md]: missing `.` at the end of the sentence

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 15]
```diff
@@ -24,6 +24,7 @@ advanced users.
  - [Setting Cassandra user permissions](#Setting_Cassandra_user_permissions)
  - [Cassandra table level configuration](#Cassandra_table_level_configuration) 
  - [Mail Queue](#Mail_Queue)
+ - [Deleted Messages Vault](#Deleted_Messages_Vault)
```
Comment (mbaechler) [inline src/site/markdown/server/manage-guice-distributed-james.md]: Any specific order here?

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 16]
```diff
@@ -493,3 +494,38 @@ Managing an email queue is an easy task if you follow this procedure:
...
+
+In order to configure the retention time for the Deleted Messages Vault, an administrator needs to perform fine configuration tunning as explained in [deletedMessageVault.properties](https://github.com/apache/james-project/blob/master/dockerfiles/run/guice/cassandra/destination/conf/deletedMessageVault.properties). Mails are not retained forever as you have to configure a retention period (by `retentionPeriod`) before using it (with one-year retention by default if not defined)
+
+### Restore deleted messages after deletion
+
+After users deleted their mails and emptied the trash, the admin can use [Restore Deleted Messages](manage-webadmin.html#deleted-messages-vault) to restore all the deleted mails.  
+
+### Cleaning expired deleted messages
+
+You can delete all deleted messages older than the configured `retentionPeriod` by using [Purge Deleted Messages](manage-webadmin.html#deleted-messages-vault). We recommend calling this API in CRON job on 1st day each month.
```
Comment (mbaechler) [inline src/site/markdown/server/manage-guice-distributed-james.md]: Before reaching this point, we don't know we _have to_ put that CRON in place if we want retention to be applied.

It's a very important information and it should be promoted to the top section.

File: src/site/markdown/server/manage-guice-distributed-james.md (hunk 12 above)
Comment (naviat) [reply src/site/markdown/server/manage-guice-distributed-james.md]: I'm not saying that :trollface:

File: src/site/markdown/server/manage-guice-distributed-james.md (hunk 16 above)
Comment (naviat) [reply src/site/markdown/server/manage-guice-distributed-james.md]: Pushed it to the top of this Section. Done

File: src/site/markdown/server/manage-guice-distributed-james.md [hunk 17]
```diff
@@ -505,7 +505,7 @@ Deleted Messages Vault is an interesting feature that will help James users have
 
 If the Deleted Messages Vault is enabled when users delete their mails, and by that we mean when they try to definitely delete them by emptying the trash, James will retain these mails into the Deleted Messages Vault, before an email or a mailbox is going to be deleted. And only administrators can interact with this component via [WebAdmin REST APIs](manage-webadmin.html#deleted-messages-vault).
 
-However, mails are not retained forever as you have to configure a retention period before using it (with one-year retention by default if not defined). It's also possible to permanently delete a mail if needed.
+However, mails are not retained forever as you have to configure a retention period before using it (with one-year retention by default if not defined). It's also possible to permanently delete a mail if needed and we recommend administrator need to [run it](#Cleaning_expired_deleted_messages) in cron job to save storage volume.
```
Comment (Arsnael) [inline src/site/markdown/server/manage-guice-distributed-james.md]: `we recommend the administrator to`

**Comment by naviat** [comment]
@mbaechler  @chibenwa  Do we need any resolve with this?

**Comment by chibenwa** [comment]
> Needs a rebase though sorry :)

Easy to do upon merge, let's not anoy them

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**